pkg/virtual/initializingworkspaces/builder/build.go

```diff
@@ -0,0 +1,241 @@
+			if initializerName == "" {
+				return
+			}
+
+			realPath := "/" + parts[1]
+
+			//  /services/initializingworkspaces/<initializer>/clusters/*/apis/workload.kcp.dev/v1alpha1/workloadclusters
+			//                  ┌─────────────────────────────┘
+			// We are now here: ┘
+			// Now, we parse out the logical cluster and validate that only wildcard requests are allowed.
+			cluster := genericapirequest.Cluster{Name: logicalcluster.Wildcard, Wildcard: true}
+			if strings.HasPrefix(realPath, "/clusters/") {
+				withoutClustersPrefix := strings.TrimPrefix(realPath, "/clusters/")
+				parts = strings.SplitN(withoutClustersPrefix, "/", 2)
+				lclusterName := parts[0]
+				realPath = "/"
+				if len(parts) > 1 {
+					realPath += parts[1]
+				}
+				if lclusterName != "*" {
+					return false, "", nil
+					// TODO: how do we signal that this is an error?
+				}
+			}
+
+			completedContext = genericapirequest.WithCluster(requestContext, cluster)
+			completedContext = dynamiccontext.WithAPIDomainKey(completedContext, dynamiccontext.APIDomainKey(initializerName))
+			prefixToStrip = strings.TrimSuffix(urlPath, realPath)
+			accepted = true
+			return
+		},
+		Ready: func() error {
+			select {
+			case <-readyCh:
+				return nil
+			default:
+				return errors.New("initializingworkspaces virtual workspace controllers are not started")
+			}
+		},
+		BootstrapAPISetManagement: func(mainConfig genericapiserver.CompletedConfig) (apidefinition.APIDefinitionSetGetter, error) {
+			retriever := &apiSetRetriever{
+				config:               mainConfig,
+				dynamicClusterClient: dynamicClusterClient,
+				crdInformer:          wildcardApiExtensionsInformers.Apiextensions().V1().CustomResourceDefinitions(),
+			}
+
+			if err := mainConfig.AddPostStartHook(initializingworkspaces.VirtualWorkspaceName, func(hookContext genericapiserver.PostStartHookContext) error {
+				defer close(readyCh)
+
+				for name, informer := range map[string]cache.SharedIndexInformer{
+					"customresourcedefinitions": wildcardApiExtensionsInformers.Apiextensions().V1().CustomResourceDefinitions().Informer(),
+				} {
+					if !cache.WaitForNamedCacheSync(name, hookContext.StopCh, informer.HasSynced) {
+						return errors.New("informer not synced")
+					}
+				}
+
+				return nil
+			}); err != nil {
+				return nil, err
+			}
+
+			return retriever, nil
+		},
+	}
+}
+
+type apiSetRetriever struct {
+	config               genericapiserver.CompletedConfig
+	dynamicClusterClient dynamic.ClusterInterface
+	crdInformer          crdinfomer.CustomResourceDefinitionInformer
+}
+
+func (a *apiSetRetriever) GetAPIDefinitionSet(ctx context.Context, key dynamiccontext.APIDomainKey) (apis apidefinition.APIDefinitionSet, apisExist bool, err error) {
+	initializerName := string(key)
+
+	crds, err := a.crdInformer.Lister().List(labels.Everything())
+	if err != nil {
+		return nil, false, err
+	}
+	for _, crd := range crds {
+		fmt.Printf("CRD: %#v\n", crd)
+	}
```

Comment:
this is debug code?

(no need to change this right away, we will come back here anyway) wildcard informer is far too much. We watch exactly one CRD here, in one workspace, known by name.